src/lakefs_spec/client_helpers.py

```diff
@@ -31,59 +32,88 @@ def commit(
         logger.warning(f"No changes to commit on branch {branch!r}, aborting commit.")
         return rev_parse(client, repository, branch, parent=0)
 
-    commit_creation = CommitCreation(message=message, metadata=metadata or {})
+    commit_creation = CommitCreation(message=message, metadata=metadata)
 
     new_commit = client.commits_api.commit(
         repository=repository, branch=branch, commit_creation=commit_creation
     )
     return new_commit
 
 
-def create_tag(client: LakeFSClient, repository: str, ref: str | Commit, tag: str) -> None:
-    if isinstance(ref, Commit):
-        ref = ref.id
-    tag_creation = TagCreation(id=tag, ref=ref)
-    client.tags_api.create_tag(repository=repository, tag_creation=tag_creation)
-
-
-def ensure_branch(client: LakeFSClient, repository: str, branch: str, source_branch: str) -> str:
+def create_branch(
+    client: LakeFSClient, repository: str, name: str, source_branch: str, exist_ok: bool = True
+) -> str:
     """
-    Creates a branch named ``branch`` if not already existent.
+    Create a branch in a lakeFS repository.
 
     Parameters
     ----------
     client: LakeFSClient
         The lakeFS client configured for (and authenticated with) the target instance.
     repository: str
         Repository name.
-    branch: str
-        Name of the branch.
+    name: str
+        Name of the newly created branch.
     source_branch: str
         Name of the source branch the new branch is created from.
+    exist_ok: bool
+        Ignore errors if the branch already exists.
 
     Returns
     -------
-    The branch name that was given.
+    The newly created branch name.
```

Comment:
Reword to include the case it already exists

Reply:
Addressed in [643d821](https://github.com/aai-institute/lakefs-spec/pull/154/commits/643d821ad647f440a4dca82841dcbcd65d9c79b7).